utils/kayobe-automation-redact

```diff
@@ -5,11 +5,21 @@ import hashlib
 import base64
 import sys
 
+annotation_exceptions = {
+    'prometheus_bcrypt_salt': {'original': 'prometheusbcryptsalt.o', 'changed': 'prometheusbcryptsalt.c'},
+}
+
 def annotate(ctx, value):
-  if not isinstance(value, str):
-    return value
-  path_str = map(str, ctx['path'])
-  return f"{'.'.join(path_str)}.{ value }"
+    if not isinstance(value, str):
+        return value
+    path_str = ctx['path'][0]
```

Comment:
I'm not exactly sure what the path is here, but we've lost anything after the first part of it here, whereas previously it was joining with `.`.

Reply:
ctx['path'] contains a list with a single element which is the line from file being redacted.

Something like this.

```py
['prometheus_bcrypt_salt']
```

My change simply extracts the first and only element and does away with the `map` where `.join` was being used to expand the `map` object.

Original approach

```yaml
---
bar_baz: bar_baz.original
foo_bar: foo_bar.original
prometheus_bcrypt_salt: prometheus_bcrypt_salt.original
```

Fixed approach for bcrypt_salt

```yaml
---
bar_baz: bar_baz.original
foo_bar: foo_bar.original
prometheus_bcrypt_salt: prometheusbcryptsalt.o
```

Reply:
Are you sure that there will never be other items in the path list? It looks like @jovial at least considered that case. I'll defer to him.

Reply:
I thought the ssh keys were nested dicts:
```
bifrost_ssh_key:
  private_key: value
  public_key: value
 ```
 so path be something like:
 ```
 ['bifrost_ssh_key', 'private_key']
 ```
 just makes it a bit ambiguous where it came from if you only use `private_key`

Reply:
annotation_exceptions seems like a reasonable approach though. Could keep the existing behaviour for the non-exception case?